Source/WarpX.H

```diff
@@ -113,11 +114,27 @@ public:
 
     void Evolve (int numsteps = -1);
 
-    void EvolveImplicitPicardInit (int lev);
-    void SaveParticlesAtImplicitStepStart (WarpXParticleContainer& pc, int lev);
-    void FinishImplicitParticleUpdate (WarpXParticleContainer& pc, int lev);
-    void FinishImplicitFieldUpdate(amrex::Vector<std::array< std::unique_ptr<amrex::MultiFab>, 3 > >& Efield_fp,
-                                   amrex::Vector<std::array< std::unique_ptr<amrex::MultiFab>, 3 > >& Efield_n);
+    //
+    // Functions used by implicit solvers
+    //
+    void PreRHSOp ( amrex::Real  cur_time,
+                    amrex::Real  a_full_dt,
+                    int          a_nl_iter,
+                    bool         a_from_jacobian );
+    void SaveParticlesAtImplicitStepStart ();
+    void FinishImplicitParticleUpdate ();
+    void SetElectricFieldAndApplyBCs ( const WarpXSolverVec&  a_E );
+    void UpdateMagneticFieldAndApplyBCs ( const amrex::Vector<std::array< std::unique_ptr<amrex::MultiFab>, 3 > >& a_Bn,
+                               amrex::Real a_thetadt );
+    void ApplyMagneticFieldBCs ();
+    void FinishMagneticFieldAndApplyBCs ( const amrex::Vector<std::array< std::unique_ptr<amrex::MultiFab>, 3 > >& a_Bn,
+                               amrex::Real a_theta );
+    void FinishImplicitField ( amrex::Vector<std::array< std::unique_ptr<amrex::MultiFab>, 3 > >& Field_fp,
+                         const amrex::Vector<std::array< std::unique_ptr<amrex::MultiFab>, 3 > >& Field_n,
+                         amrex::Real theta );
+    void ComputeRHSE (         amrex::Real dt, WarpXSolverVec& a_Erhs_vec);
+    void ComputeRHSE (int lev, amrex::Real dt, WarpXSolverVec& a_Erhs_vec);
+    void ComputeRHSE (int lev, PatchType patch_type, amrex::Real dt, WarpXSolverVec& a_Erhs_vec);
```

Comment:
Would it make sense to add `implicit` in the name of this function, so that other developers can more easy identify that this function relates to the implicit algorithm?
```suggestion
    void ImplicitComputeRHSE (         amrex::Real dt, WarpXSolverVec& a_Erhs_vec);
    void ImplicitComputeRHSE (int lev, amrex::Real dt, WarpXSolverVec& a_Erhs_vec);
    void ImplicitComputeRHSE (int lev, PatchType patch_type, amrex::Real dt, WarpXSolverVec& a_Erhs_vec);
```

Reply:
Agreed.